install.ps1

```diff
@@ -71,21 +72,25 @@ run the script and if it still fail please open an issue on the Dagger repo.
 
 function base_url {
     if ($DaggerVersion) {
-        $version = $DaggerVersion
+        $path = "releases/" + $DaggerVersion
+    } elseif ($DaggerCommit) {
+        $path = "main/" + $DaggerCommit
     } else {
-        $version = latest_version
+        $path = "releases/" + (latest_version)
     }
-    $url = $base + "/" + $name + "/releases/" + $version
+    $url = $base + "/" + $name + "/" + $path
     return $url
 }
 
 function tarball {
     if ($DaggerVersion) {
-        $version = $DaggerVersion
+        $version = "v" + $DaggerVersion
+    } elseif ($DaggerCommit) {
+        $version = $DaggerCommit
     } else {
-        $version = latest_version
+        $version = "v" + (latest_version)
     }
-    $fileName="dagger_v" + $version + "_windows_amd64"
+    $fileName="dagger_" + $version + "_windows_amd64"
```

Comment:
I thought this PR was only for .sh but can you check my PR maybe for some improvements i.e:

- The amd64 is fixed here but Windows runs ARM too WinRT
- The $InteractiveInstall could be a `[switch]` for a cleaner looking `.\script -Install` (no bool)

Reply:
Thanks @pjmagee for #7569, they look pretty good to me!

I think both PRs do something different, this is just adding a new feature into both scripts to make it easier to consume builds from `main`. I think my idea is to try and merge this one first, and then we can rebase #7569 on top of that with changes?

Reply:
yeah, Lets do that

Reply:
The keeping it inline was mostly just me attempting to tidy up the windows script when I reimported it - there's no issue if you want to diverge IMO, if that's more familiar to windows users :tada: